torch/autograd/profiler.py

```diff
@@ -233,17 +240,19 @@ def __init__(
                 use_kineto
             ), "Device-only events supported only with Kineto (use_kineto=True)"
 
-        if self.use_device == "cuda":
-            self.use_device = None
-            self.use_cuda = True
-
-        if self.use_device and self.use_device != _get_privateuse1_backend_name():
-            warn(f"{self.use_device} doesn't support profile.")
+        VALID_DEVICE_OPTIONS = ["cuda", "xpu", "privateuseone"]
+        if self.use_device not in VALID_DEVICE_OPTIONS:
```

Comment:
This results in an annoying `UserWarning: The None is not a valid device option.` warning whenever CPU profiler is used :( PR is coming

Reply:
Thank you @malfet